File: reference/array-hash/1-Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_sum = {}
        for i in range(len(nums)):
            result = target - nums[i]
            if result in hash_sum:
                return [i, hash_sum[result]]
            hash_sum[nums[i]] = i
        return []

File: reference/array-hash/1-Two-Sum.go
func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)

    for i, num := range nums {
        complement := target - num
        if idx, found := hashMap[complement]; found {
            return []int{idx, i}
        }
        hashMap[num] = i
    }

    return nil
}